src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -186,7 +187,15 @@ public function handle(Request $request, $type = HttpKernelInterface::MASTER_REQ
             $this->boot();
         }
 
-        return $this->getHttpKernel()->handle($request, $type, $catch);
+        try {
+            if (0 === $this->requestStackSize++ && $this->container->has('services_resetter')) {
```

Comment:
Wouldn't it be sufficient to check for a master request?

Reply:
Sufficient like calling reset() in kernel.terminate ? ;)
I'm sure there is code out there that would thank us being extra cautious here.